### PR TITLE
Fix for when date is passed as 'date' header.

### PR DESCRIPTION
**Pull request by ghost**
- Use method for getting request date rather than call x-amz-date key
  from headers in AWS4Auth get_sig_string.

### CODE REVIEW

**Comment by leepa** [comment]
You need to support both though... as most libraries us the other header.

**Comment by ghost** [comment]
they are both supported, the lib checks to see if either exists.

**Comment by stevenh** [comment]
I don't think this is the right place for this as its working on a prepared request not a request so its correct for it to get from the header it knows is there (assuming it was prepared correctly).

File: requests_aws4auth/aws4auth.py [hunk 1]
```diff
@@ -583,7 +584,11 @@ def get_sig_string(req, cano_req, scope):
                     get_canonical_request()
 
         """
-        amz_date = req.headers['x-amz-date']
+        if 'x-amz-date' in req.headers:
```
Comment (stevenh) [inline requests_aws4auth/aws4auth.py:590]: Method doc string needs updating to reflect this change.

File: requests_aws4auth/aws4auth.py [hunk 2]
```diff
@@ -577,13 +578,20 @@ def get_sig_string(req, cano_req, scope):
         """
         Generate the AWS4 auth string to sign for the request.
 
+        Checks to see if date has been passed as x-amz-date or date, if
+        date then parse the date from an RFC 7231 string.
```
Comment (stevenh) [inline requests_aws4auth/aws4auth.py]: if only date is present.

email.utils doesn't seems to mention RFC 7231 so are you sure it does?

File: requests_aws4auth/aws4auth.py [hunk 3]
```diff
@@ -577,13 +578,20 @@ def get_sig_string(req, cano_req, scope):
...
+
         req      -- Requests PreparedRequest object. This should already
-                    include an x-amz-date header.
+                    include an x-amz-date, or a date header.
```
Comment (stevenh) [inline requests_aws4auth/aws4auth.py]: no need for the comma

**Comment by stevenh** [comment]
:+1:

**Pull request merged**